src/api/gold_prices/get_gold_prices.rs

```diff
@@ -0,0 +1,69 @@
+use crate::client::service_client::ServiceClient;
+use crate::models::date_parameters::{DateParameter, NoDateParameter, WithDateParameter};
+use crate::models::gold_prices::GoldPrice;
+use crate::nbp_error::NbpResult;
+use chrono::NaiveDate;
+use std::marker::PhantomData;
+
+pub struct GetGoldPricesBuilder<State> {
+    service_client: ServiceClient,
+    date_parameter: Option<DateParameter>,
+    _state: PhantomData<State>,
+}
+
+impl GetGoldPricesBuilder<NoDateParameter> {
+    pub fn new(service_client: ServiceClient) -> Self {
+        Self {
+            service_client,
+            date_parameter: None,
+            _state: PhantomData,
+        }
+    }
+
+    pub fn today(self) -> GetGoldPricesBuilder<WithDateParameter> {
+        GetGoldPricesBuilder {
+            service_client: self.service_client,
+            date_parameter: Some(DateParameter::Today),
+            _state: PhantomData,
+        }
```

Comment:
We could consider using the struct update syntax to reduce code duplication and improve clarity as to what is changing:
```
GetGoldPricesBuilder {
  date_parameter: Some(DateParameter::Today),
  ..self
}
```

Reply:
This comment applies to all the builder methods.

Reply:
The `..self` syntax only works if the returned `State` matches the initial state of the `Builder`. If `new()` initializes the state as `NoDateParameter` but the method tries to return `WithDateParameter`, `..self` won't work because the types differ.  

Instead, you must explicitly construct a new `Builder` with the updated state, as Rust enforces type consistency in struct updates.

Reply:
Makes sense, I must have overlooked that aspect. Thanks!